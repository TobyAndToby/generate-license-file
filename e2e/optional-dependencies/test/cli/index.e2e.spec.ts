import { exec } from "child_process";
import fs from "fs/promises";
import { promisify } from "util";

const execAsync = promisify(exec);

describe("cli", () => {
  let output = "";

  beforeEach(async () => {
    output = `./test-${Math.floor(Math.random() * 1000)}.txt`;
  });

  afterEach(async () => {
    await fs.unlink(output);
  });

  it("should match snapshot", async () => {
    await execAsync(
      `npx generate-license-file --input ./package.json --output ${output}`
    );

    const result = await fs.readFile(output, "utf8");
    expect(result).toMatchSnapshot();
  });
});
