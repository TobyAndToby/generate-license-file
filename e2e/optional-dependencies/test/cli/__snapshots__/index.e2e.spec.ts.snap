// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should match snapshot 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;
